simplesat/repository.py

```diff
@@ -75,7 +75,7 @@ def add_package(self, package_metadata):
         If the same package is added multiple times to a repository, every copy
         will be available when calling find_package or when iterating.
         """
-        if package_metadata.name not in self._name_to_packages:
+        if len(self._name_to_packages[package_metadata.name]) == 0:
```

Comment:
So the intention here *is* to insert new keys at this point and make it such that it makes no difference whether a key is empty or missing?  Is that correct?